pkg/status/status.go

```diff
@@ -191,12 +259,90 @@ func (s *status) setStatusCondition(condition configv1.ClusterStatusConditionTyp
 		Message:            failingMessage,
 		LastTransitionTime: time,
 	})
+	return updatedCondition
 }
 
-// updateStatus makes the API call to update the ClusterOperator
-func (s *status) updateStatus() {
-	_, err := s.configClient.ClusterOperators().UpdateStatus(s.clusterOperator)
-	if err != nil {
-		log.Fatalf("Error %v updating ClusterOperator", err)
+// updateStatus makes the API call to update the ClusterOperator if the status has changed.
+func (s *status) updateStatus(previousStatus *configv1.ClusterOperatorStatus, updatedCondition string, message string) {
+	if reflect.DeepEqual(previousStatus, &s.clusterOperator.Status) {
+		log.Debugf("Previous and current clusterOperator Status are the same, the clusterOperator Status will not be updated.")
+	} else {
+		log.Debugf("Previous and current clusterOperator Status are different, attempting to update the clusterOperator Status.")
+
+		// Check if the clusterOperator version has changed and log the attempt to upgrade if it has
+		previousVersion := operatorhelpers.FindOperandVersion(previousStatus.Versions, "operator")
+		currentVersion := operatorhelpers.FindOperandVersion(s.clusterOperator.Status.Versions, "operator")
+		if currentVersion != nil {
+			if previousVersion == nil {
+				log.Infof("Attempting to upgrade clusterOperator to version %s", currentVersion.Version)
```

Comment:
Why would this be an upgrade if the `previousVersion` was `nil`?

Reply:
Good point. I will change this from "Attempting to upgrade..." to "Attempting to set..."